### ISSUE TITLE
Issue with Routes

### ISSUE THREAD

**Issue by prabhuomkar** [siesgstarena/Arena-2.0#1]
Routes are by default redirected to:
`https://siesgstarena.github.io/{routeName}`

**Issue**: Reloading the webpage shows an error as all routes should be `https://siesgstarena.github.io/Arena-2.0/{routeName}`

**Fix**: Make the base endpoint to be:
`https://siesgstarena.github.io/Arena-2.0/`

Also, name all the routes (except static routes) like `/{service}/{method}` e.g.:
/auth/signup
/auth/login
/contests/:code
/profile/:id
/profile/settings

**Comment by ac030540** [comment]
I think the issue with routes you are talking about is related to hosting on github pages.
By default '/' is taken as home (base address) and all the routes come up after that e.g. '/{route_name}'. But when it is deployed on github pages, the base address becomes '/Arena-2.0' and hence reloading issue is there. I will update those routes and also will name them according to the convention you suggested.

**Comment by prabhuomkar** [comment]
Make it switchable for prod and github versions so that only GitHub version can use it as `/Arena-2.0/`

Close it, when you fix it on next commit?

**Comment by ac030540** [comment]
Okay. I will do that.

**Comment by ac030540** [comment]
Just to inform, I tried to fix the issue by updating the making all the routes to `https://siesgstarena.github.io/Arena-2.0/{routeName}` but still reloading doesn't works. From what I have read, I think the issue can be  because when we reload a page (when app is deployed on github pages), it looks for the page.html file in the our app. But since in react we make always have a single HTML file, github pages cannot file the particular file and gives an error. I got some resources to fix the issue and I am working on it.
